src/main/react/app/containers/App/excludedDate/index.js

```diff
@@ -1,208 +1,121 @@
 import React from "react";
-import DateInput from "./dateInput";
-import { DATE_TYPE, getDatePresets, PERIODS} from "../constants";
-import {
-  formatDate,
-  nextOccurrenceChineseLunar
-} from "../../../utils/date";
-import { getBrief } from "../../../utils";
-import moment from "moment";
-import {
-  RepeatCount,
-  RepeatInterval,
-  RepeatPeriod,
-  NameInput,
-  TimezoneInput,
-  PresetSelect,
-  RepeatCheckbox
-} from "./formItems";
+import {DATE_TYPE, getDatePresets, PERIODS} from "../constants";
 
+import ExcludedDate from './excludedDate'
+import {getExcludedDates, setExcludedDates} from "../../../api";
+import only from "only";
 
-export default class ExcludeDate extends React.Component {
+export default class ExcludeDateContainer extends React.Component {
   constructor() {
     super();
     this.state = {
-      selectedDateType: DATE_TYPE.TYPE_GREGORIAN,
-      selectedPreset: -1,
-      nextOccurrence: undefined,
-
-      name: "",
-      type: "",
-
-      utcOffset: moment().utcOffset(),
-      timezone:"",
-
-      startDate: {
-        dynamic: false,
-        date: new Date(),
-        dynamicDate: {
-          month: 1,
-          week: 1,
-          day: 1
-        }
-      },
-      endDate: {
-        dynamic: false, date: new Date(), dynamicDate: {
-          month: 1,
-          week: 1,
-          day: 1
-        }
-      },
-      timezones: [],
-      noEnd: true, //No end in repeat
-      repeat: true,
-      repeatCount: -1,
-      repeatInterval: 1,
-      repeatPeriod: PERIODS.Year
+      excludedDates: [],
+      openIndex: -1,
     };
   }
 
-  /*Apply the selected preset.*/
-  applyPreset = () => {
-    /*According to the selected date type, use different algorithms.*/
-    switch (this.state.selectedDateType) {
-      case DATE_TYPE.TYPE_GREGORIAN:
-        this.setState(this.getDatePresets()[this.state.selectedPreset]);
-        break;
-      case DATE_TYPE.TYPE_CHINESE_LUNAR:
-        let preset = this.getDatePresets()[this.state.selectedPreset];
-        this.setState(preset);
-
-        /*For types that are not gregorian, we calculate its occurrence
-        * using its corresponding algorithm.*/
-        this.setState({
-          nextOccurrence: nextOccurrenceChineseLunar(preset.params)
-        });
-    }
-  };
-
-  handleRepeatChange = (event) => {
-    this.setState({ repeat: event.target.checked });
-  };
+  /**
+   * Handler for changing a excluded date
+   * @param index The index of the child
+   * @param show Status that indicates whether the child want to be unfold or fold
+   * @param state The new state.
```

Comment:
I like this style of comment - eventually I'd like to see it some of the other js files :-)